drivers/sdhc/sdhci_common.c

```diff
@@ -695,14 +809,32 @@ static int sdhci_check_version(struct sdhci_common *sdhci_ctx)
 	return 0;
 }
 
-int sdhci_enable_dma(struct sdhci_common *sdhci_ctx, enum sdhc_dma_select dma_mode)
+int sdhci_enable_sdma(struct sdhci_common *sdhci_ctx)
 {
-	if (dma_mode != SDHC_DMA_SDMA) {
-		return -ENOTSUP;
+	sdhci_ctx->f_use_dma = true;
+	sdhci_ctx->dma_mode = SDHC_DMA_SDMA;
+
+	return 0;
+}
+
+int sdhci_enable_adma2(struct sdhci_common *sdhci_ctx, struct sdhc_adma_desc *descs_table,
+		       uint32_t descs_table_size)
+{
+	if (!descs_table || !descs_table_size) {
+		return -EINVAL;
 	}
 
+#if defined(CONFIG_64BIT)
+	/* Only 32bit DMA supported */
+	if (((uint64_t)descs_table >> 32)) {
```

Comment:
Is it possible that the DMA address is 64 bits and the virtual address is 32 bits for this descriptor table? If yes, could you please check the DMA address here instead of the virtual one?

And isn't it enough to add some print and to disable ADMA2 support without panicking ?

Reply:
It's very unsafe and not smth expected by Zephyr where max features configured during init. User can switch for PIO mode now for 64bit xfer or anybody who need can add 64bit implementation in the future.